### ISSUE TITLE
ifName plugin default tag value

### ISSUE THREAD

**Issue by Hipska** [influxdata/telegraf#8011]
## Feature Request

Opening a feature request kicks off a discussion.
As discussed in https://github.com/influxdata/telegraf/pull/7763#issuecomment-676801651

### Proposal:

Let the default value of `tag` for `processors.ifname` be compatible with `inputs.snmp` (agent_host) or `inputs.sflow` (agent_address). 

### Current behavior:

The default value (agent) mostly never useful.

### Desired behavior:

The default value should be a logical one that makes sense or most users would not need to change.

**Comment by Hipska** [comment]
@reimda what do you think?

**Comment by Hipska** [comment]
Ping @ssoroka @reimda

**Comment by reimda** [comment]
I agree that the tags across these three plugins should match.  It was an oversight from the beginning that they don't match.

We haven't been able to change the default because it will break anyone who is using the default.  We can change the default on a major release (2.0) because major releases won't guarantee settings compatibility.

For now we ought to prepare for a change in 2.0 by choosing the tag we want to use across all three plugins and deprecating the defaults in all three plugins that don't match the tag we chose.

**Comment by Hipska** [comment]
According to #4413, `source` would be the most logical default value.

**Comment by srebhan** [comment]
@Hipska I think the `source` tag is already taken e.g. in `inputs.snmp`. Maybe we should **add** an additional `agent` tag for the inputs so you can `tag_exclude` the old one if you wish but downstream queries still work!?

**Comment by Hipska** [comment]
No, not by default. Only if you added a OID as tag and gave it that name.

**Comment by srebhan** [comment]
To which of they points is your comment? ;-)

**Comment by Hipska** [comment]
The first one, `inputs.snmp` does not have a `source` tag, it is using `agent_host` as explained in my initial post.

**Comment by powersj** [comment]
next steps: look at adding the `source` tag to the above three plugins in addition to the existing tags. Then determine if/when what minor release we could remove the old ones with documentation as to how to use the old tag name if users wanted to.

**Comment by Hipska** [comment]
@powersj I'm not sure how to deprecate the agent tag default value for the ifname processor. Did the other 2 already..
